examples/generative/gan_ada.py

```diff
@@ -0,0 +1,729 @@
+* **[residual connections](https://keras.io/guides/functional_api/#a-toy-resnet-model)**:
+While residual discriminators behave similarly, residual generators are more difficult to
+train in my experience. They are however necessary for training large and deep
+architectures. I recommend starting with non-resiudal architectures.
+* **dropout**: Using dropout before the last layer of the discriminator improves
+generation quality in my experience. Recommended dropout rate is below 0.5.
+* **[leaky ReLU](https://keras.io/api/layers/activation_layers/leaky_relu/)**: Use leaky
+ReLU activations in the discriminator to make its gradients less sparse. Recommended
+slope/alpha is 0.2 following DCGAN.
+
+### Algorithmic tips
+* **loss functions**: Numerous losses have been proposed over the years for training
+GANs, promising improved performance and stability. I have implemented 5 of them in
+[this repository](https://github.com/beresandras/gan-flavours-keras), and my experience is in
+line with [this GAN study](https://arxiv.org/abs/1711.10337): no loss seems to
+consistently outperform the default non-saturating GAN loss. I recommend using that as a
+default.
+* **Adam's beta_1 parameter**: The beta_1 parameter in Adam can be interpreted as the
+momentum of mean gradient estimation. Using 0.5 or even 0.0 instead of the default 0.9
+value was proposed in DCGAN and is important. This example would not work using its
+default value.
+* **separate batch normalization for generated and real images**: The forward pass of the
+discriminator should be separate for the generated and real images. Doing otherwise can
+lead to artifacts (45 degree stripes in my case) and decreased performance.
+* **exponential moving average of generator's weights**: This helps to reduce the
+variance of the KID measurement, and helps in averaging out the rapid color palette
+changes during training.
+* **[different learning rate for generator and discriminator](https://arxiv.org/abs/1706.08500)**:
+If one has the resources, it can help
+to tune the learning rates of the two networks separately. A similar idea is to update
+either network's (usually the discriminator's) weights multiple times for each of the
+other network's updates. I recommend using the same learning rate of 2e-4 (Adam),
+following DCGAN for both networks as a default.
+* **label noise**: [One-sided label smoothing](https://arxiv.org/abs/1606.03498) (using
+less than 1.0 for real labels), or adding noise to the labels can regularize the
+discriminator not to get overconfident, however in my case they did not improve
+performance.
+* **adaptive data augmentation**: Since it adds another dynamic component to the training
+process, disable it as a default, and only enable it when the other components already
+work well.
+"""
+
+"""
+## Related works
+
+Other GAN-related Keras code examples:
+* [DCGAN + CelebA](https://keras.io/examples/generative/dcgan_overriding_train_step/)
```

Comment:
Add line break before list start (same for the other list below). This is necessary for rendering: https://github.com/keras-team/keras-io/blob/master/contributor_guide.md#markdown-lists